internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -237,82 +443,486 @@ func (s *Server) HandleGitHubWebHook() http.HandlerFunc {
+		}
+
+		if processingErr != nil {
+			wes = handleParseError(wes.Typ, processingErr)
 			if wes.Error {
 				w.WriteHeader(http.StatusInternalServerError)
-			} else {
-				w.WriteHeader(http.StatusOK)
+				return
 			}
+
+			w.WriteHeader(http.StatusOK)
 			return
 		}
-		wes.Accepted = true
-		l.Info().Str("message-id", m.UUID).Msg("publishing event for execution")
 
-		// Channel the event based on the webhook action
-		var watermillTopic string
-		if shouldIssueDeletionEvent(m) {
-			watermillTopic = events.TopicQueueReconcileEntityDelete
-		} else {
-			watermillTopic = events.TopicQueueEntityEvaluate
-		}
+		// res is null only when a ping event occurred.
+		if res != nil && res.eiw != nil {
+			if err := res.eiw.ToMessage(m); err != nil {
+				wes.Error = true
+				log.Printf("Error creating event: %v", err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+
+			// Publish the message to the event router
+			if err := s.evt.Publish(res.topic, m); err != nil {
+				wes.Error = true
+				log.Printf("Error publishing message: %v", err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}
+
+		// We successfully published the message
+		wes.Error = false
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+// processPingEvent logs the type of token used to authenticate the
+// webhook. The idea is to log a link between the repo and the token
+// type. Since this is done only for the ping event, we can assume
+// that the sender is the app that installed the webhook on the
+// repository.
+func (_ *Server) processPingEvent(
+	ctx context.Context,
+	payload []byte,
+) {
+	l := zerolog.Ctx(ctx).With().Logger()
+
+	var event *pingEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		l.Info().Err(err).Msg("received malformed ping event")
+		return
+	}
+
+	if event.GetRepo() != nil {
+		l = l.With().Int64("github-repository-id", event.GetRepo().GetID()).Logger()
+		l = l.With().Str("github-repository-url", event.GetRepo().GetHTMLURL()).Logger()
+	}
+	if event.GetSender() != nil {
+		l = l.With().Str("sender-login", event.GetSender().GetLogin()).Logger()
+		l = l.With().Str("github-repository-url", event.GetSender().GetHTMLURL()).Logger()
+		if strings.Contains(event.GetSender().GetHTMLURL(), "github.com/apps") {
+			l = l.With().Str("sender-token-type", "github-app").Logger()
+		} else {
+			l = l.With().Str("sender-token-type", "oauth-app").Logger()
+		}
+	}
+
+	l.Debug().Msg("ping received")
+}
+
+func (s *Server) processPackageEvent(
+	ctx context.Context,
+	payload []byte,
+) (*processingResult, error) {
+	var event *packageEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return nil, err
+	}
+
+	if event.Action == nil {
+		return nil, errors.New("invalid event: action is nil")
+	}
+	if event.Package == nil || event.Repo == nil {
+		log.Printf("could not determine relevant entity for event. Skipping execution.")
+		return nil, nil // this is awkward
```

Comment:
I agree, feel free to return e.g. `errNotHandled` or introduce a new error that would be handled in `handleParseError`